### ISSUE TITLE
AWG in continious mode

### ISSUE THREAD

**Issue by LRSchreiber** [qutech/qupulse#441]
Implement an option for continious running of AWG pulses without the need of a trigger signal

**Comment by terrorfisch** [comment]
So you mean an inifinite repetition of the currently active pulse? Can you specify how you would use this and what you expect of the device? What would you expect if a pulse is uploaded on multiple devices by the HardwareBackend?

I think this can currently be hacked manually, at least with the tabor AWG in the short term. I put this into the features of the new AWG model.

**Comment by LRSchreiber** [comment]
yes I mean infinite repetition of the currently active pulse, which might contail subsequences. I expect that the pulse can be uploaded. Then started and stopped by a software command. Between start and stop command the pulse runs continuously. If such a pulse is uploaded on multiple devices synchronoty between pulses is not required. The normal mode of a cont. mode is basic testing using one AWG only, but potentially all channels and marker channels of an AWG. I am sure this can be hacked somehow on the short term.

**Comment by terrorfisch** [comment]
I implemented an interface for this feature in #464 for the Tektronix driver:
```python
    @property
    def default_program_repetition_mode(self) -> str:
        """repetition mode for newly uploaded programs. Valid values are 'once' and 'infinite'. You can use
        set_program_repetition_mode to change the repetition mode of an existing program"""
        return self._default_program_repetition_mode

    @default_program_repetition_mode.setter
    def default_program_repetition_mode(self, mode: str):
        self._validate_program_repetition_code(mode)
        self._default_program_repetition_mode = mode

    def set_program_repetition_mode(self, program_name: str, mode: str):
        """instrument specific"""

    def get_program_repetition_mode(self, program_name: str) -> str:
        """instrument specific"""
```

If this turns out to be a feasible way of doing it this should become part of the AWG interface class.
